ha/setup/post_disruptive_upgrade.py

```diff
@@ -0,0 +1,163 @@
+    new_conf_dict = _yaml_to_dict(ha_source_conf)
+
+    # Note: There are 3 scenarios for conf file upgrade
+    # 1. New conf key-value pair can be introduced
+    # 2. Already present conf key can be updated with new value
+    # 3. A conf key-value can be deleted
+    # Here, we are considering or assuming that there will not be updation.
+    # Upgrade means a new key value will be added. 2nd scenario can be handled
+    # or needs to be handled seperately.
+    # If key will be deleted after upgrade and still the conf will be loaded with that
+    # key, it will not affect the functionality because, that key will not be in use.
+
+    # Update the old dictionary with new one
+    # This update will also update the values if they got changed in new
+    # version. This is not considered right now, hence update can be safely used.
+    # So, handling only first scenario here.
+    old_backup_conf_dict.update(new_conf_dict)
+
+    # Finally, update the old config file with new changes
+    with open(ha_backup_conf, 'w') as outfile:
+        yaml.dump(old_backup_conf_dict, outfile, default_flow_style=False)
+
+    try:
+        # Finally copy the updated backup conf file to a source
+        copyfile(ha_backup_conf, ha_source_conf)
+
+        # At last, copy the whole source directory which has updated
+        # conf to a desired location
+        if os.path.exists(new_src_dir) and os.listdir(new_src_dir):
+            copystat(new_src_dir, dest_dir)
+    except Exception as err:
+        raise UpgradeError('Failed to load the new config after \
+                       upgrading the RPM. Please retry Upgrade process again') \
+                       from err
+
+def _create_resources(_s3instances=None) -> None:
+    '''create required resources'''
+    create_all_resources(s3_instances=_s3instances)
+
+def _switch_cluster_mode(cluster_mode, retry_count=0) -> None:
+    '''
+       Perform cluster operation to change the mode such as standby or
+       unstandby and also retries the operation
+    '''
+    try:
+        cluster_switch_mode_command = cluster_mode + f' --wait={CLUSTER_STANDBY_UNSTANDBY_TIMEOUT}'
+        SimpleCommand().run_cmd(cluster_switch_mode_command)
+    except Exception as err:
+        if retry_count != 3:
+            retry_count += 1
+            _switch_cluster_mode(cluster_mode, retry_count)
+        raise UpgradeError('Failed to switch the mode of the cluster. \
+                            Retry upgrade again') from err
+
+def _unstandby_cluster() -> None:
+    '''Unstandby the cluster'''
+
+    _switch_cluster_mode(PCS_CLUSTER_UNSTANDBY)
+    Log.info('### cluster is up and running ###')
+
+def perform_post_upgrade(s3_instances=None):
+    '''Starting routine for post-upgrade process'''
+    Log.init(service_name="post_disruptive_upgrade", log_path=RA_LOG_DIR, level="INFO")
+    _check_for_any_resource_presence()
+    _is_cluster_standby_on()
+    _load_config()
+    _create_resources(s3_instances)
```

Comment:
Will this be called on all nodes?

Reply:
When I created the LLD for this feature, it was suppose to be for single node VM. Created LLD assuming the same fact. The previous ticket was also tested in same way.
So, I am assuming it is for single node.
Please let me know if there are any concerns

Reply:
ok